### PR TITLE
Plugin demos

### PR DESCRIPTION
**Pull request by Cr1t-GYM**
- [x] Add detailed descriptions for s3-artifact-manager demo.
- [x] Add detailed descriptions for jobcacher demo.
- [x] Add detailed descriptions for remote agent demo.
- [x] Add related changes in the main README.

### CODE REVIEW

File: demo/jobcacher/README.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
+# How to cache the pipeline data
+
+This example shows how to cache the pipeline data by using [jobcacher plugin](https://plugins.jenkins.io/jobcacher/).
+In this example, it takes `AWS S3` as an example and other compatible object storage services should work as well.
+
+If you use `AWS S3` like this example, you need to know your the region of the user, `AWS access key id`,
```
Comment (krisstern) [inline demo/jobcacher/README.md:6]:
```suggestion
If you use `AWS S3` like this example, you will need to know your the region of the user, `AWS access key id`, 
```

File: demo/remote-agent/README.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+Here is the step-by-step example in `jfr-container-action`.
+1. Firstly, follow the instructions about [generating the SSH key pair](https://www.jenkins.io/doc/book/using/using-agents/#generating-an-ssh-key-pair).
+2. Follow the example of [jenkins-secrets](../jenkins-secrets) and create two secrets for the public IPv4 address of host machine, and the SSH private key generated in the first step.
+   In this example, `AWS_SSH_HOST` corresponds to the public IPv4 address of host machine,
+   and `AWS_SSH_PKEY` corresponds to the SSH private key.
+3. Follow the instructions about [creating your Docker agent](https://www.jenkins.io/doc/book/using/using-agents/#creating-your-docker-agent). 
+   As your host machine might already have a ssh server running on the 22 port, you should use another port for the docker command, such as -p 4444:22. 
+   In this example, it uses -p 4444:22 for port mapping. 
+   If you create the machines in cloud providers such as AWS, you need to create the correct inbound rule for the port 4444.
```
Comment (krisstern) [inline demo/remote-agent/README.md:15]:
```suggestion
   If you create the machines in cloud providers such as AWS, you will need to create the correct inbound rule for the port 4444.
```

File: demo/remote-agent/README.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+   For Example, if you use port 4444 for network redirection, you need to create the TCP protocol for port 4444 and allow the network from any ip ranges.
```
Comment (krisstern) [inline demo/remote-agent/README.md:16]:
```suggestion
   For Example, if you use port 4444 for network redirection, you will need to create the TCP protocol for port 4444 and allow the network from any ip ranges.
```

File: demo/remote-agent/README.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+                maven 'maven'
+            }
+            steps {
+                sh 'mvn clean install -B --no-transfer-progress'
+            }
+        }
+    }
+}
+```
+5. In your workflow definition, you need to set up your credentials and `jfr-container-action` correctly.
```
Comment (krisstern) [inline demo/remote-agent/README.md:84]:
```suggestion
5. In your workflow definition, you will need to set up your credentials and `jfr-container-action` correctly.
```

File: demo/remote-agent/README.md (hunk 2 above)
Comment (krisstern) [inline demo/remote-agent/README.md:15]:
```suggestion
   If you create the machines in the cloud using cloud providers such as AWS, you will need to create the correct inbound rule for the port 4444.
```

File: demo/s3-artifact-manager/README.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+This example shows how to copy the pipeline results to an external object storage such as `AWS S3` by using
+[artifact-manager-s3](https://plugins.jenkins.io/artifact-manager-s3/). In this example, it takes `AWS S3` as an example
+and other object storage services should work as well.
+
+If you use `AWS S3` like this example, you need to know your the region of the user, `AWS access key id`, `AWS secret access key`, 
+`AWS S3 bucket name` and `AWS S3 bucket prefix` in advance. For example, if your `S3 URI` for a specific folder under a 
+bucket is `s3://jenkins-action/test/`, `AWS S3 bucket name` is `jenkins-action` and `AWS S3 bucket prefix` is `test/`.
+`AWS access key id` and `AWS secret access key` can be downloaded when you create a new user for your account.
+Furthermore, in order to make this AWS access key have the ability to manage your S3 storage,
+you need to give it `AmazonS3FullAccess` in the IAM user permission page.
```
Comment (krisstern) [inline demo/s3-artifact-manager/README.md:12]:
```suggestion
you will need to grant it `AmazonS3FullAccess` in the IAM user permission page.
```

File: demo/s3-artifact-manager/README.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        }
+    }
+    post {
+        success {
+            archiveArtifacts 'target/*.jar'
+        }
+    }
+}
+```
+5. In your workflow definition, you need to set up your credentials and `jfr-container-action` correctly.
```
Comment (krisstern) [inline demo/s3-artifact-manager/README.md:79]:
```suggestion
5. In your workflow definition, you will need to set up your credentials and `jfr-container-action` correctly.
```

File: demo/s3-artifact-manager/README.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+and other object storage services should work as well.
+
+If you use `AWS S3` like this example, you need to know your the region of the user, `AWS access key id`, `AWS secret access key`, 
+`AWS S3 bucket name` and `AWS S3 bucket prefix` in advance. For example, if your `S3 URI` for a specific folder under a 
+bucket is `s3://jenkins-action/test/`, `AWS S3 bucket name` is `jenkins-action` and `AWS S3 bucket prefix` is `test/`.
+`AWS access key id` and `AWS secret access key` can be downloaded when you create a new user for your account.
+Furthermore, in order to make this AWS access key have the ability to manage your S3 storage,
+you need to give it `AmazonS3FullAccess` in the IAM user permission page.
+
+Here is the step-by-step example in `jfr-container-action`.
```
Comment (krisstern) [inline demo/s3-artifact-manager/README.md:14]:
```suggestion
Here is the step-by-step example for `jfr-container-action`:
```

**Review by krisstern — CHANGES_REQUESTED** [review]
Overall the PR looks good, except some minor editing has been requested.

**Review by krisstern — APPROVED** [review]

**Review by AbhyudayaSharma — APPROVED** [review]

**Pull request merged**